jazz_deployments/test/test.js

```diff
@@ -1,11 +1,1181 @@
+        return res;
+      })
+    });
+  
+    it("should successfully initiate re-build deployment using POST method",() => {
+      event.method = "POST";
+      var responseObj = {
+        statusCode: 200,
+        body: {
+          data: {
+            token: "zaqwsxcderfv.qawsedrftg.qxderfvbhy"
+          }
+        }
+      };
+      AWS.mock("DynamoDB.DocumentClient", "query", (param, cb) => {
+        var dataObj = {
+          Items: [event.body]
+        }
+        return cb(null, dataObj);
+      });
+      reqStub = sinon.stub(request, "Request", (obj) => {
+        console.log(obj);
+        if(obj.method.toLowerCase() === 'post') {
+          return obj.callback(null, responseObj, responseObj.body)
+        } else if (obj.method === 'get') {
+          responseObj.body = "{\"data\":{\"service\":\"mag!c\",\"domain\":\"k!ngd0m\",\"type\":\"api\"},\"input\":{\"service_id\":\"k!ngd0m_0f_mewni\"}}"
+          return obj.callback(null, responseObj, responseObj.body)
+        }
+      });
+      index.handler(event, context,(err, res) => {
+          res.should.have.deep.property('data.result');
+          AWS.restore("DynamoDB.DocumentClient");
+          sinon.assert.calledThrice(reqStub);
+          reqStub.restore();
+          return res;
+      });
+      
+    });
+  
+    it("should indicate internal server error while initiating re-build deployment using POST method",() => {
+      event.method = "POST";
+      var responseObj = {
+        statusCode: 400,
+        body: {
+          data: {
+            token: "zaqwsxcderfv.qawsedrftg.qxderfvbhy"
+          }
+        }
+      };
+      AWS.mock("DynamoDB.DocumentClient", "query", (param, cb) => {
+        var dataObj = {
+          Items: [event.body]
+        }
+        return cb(null, dataObj);
+      });
+      reqStub = sinon.stub(request, "Request", (obj) => {
+        return obj.callback(null, responseObj, responseObj.body)
+      });
+      message = '{"errorType":"InternalServerError","message":"unhandled error occurred"}'
+      index.handler(event, context,(err, res) => {
+        err.should.be.equal(message);
+        sinon.assert.calledOnce(reqStub);
+        AWS.restore("DynamoDB.DocumentClient");
+        reqStub.restore();
+        return res;
+      });
+      
+    });
+  
+    it("should indicate NotFound error while initiating re-build deployment using POST method",() => {
+      event.method = "POST";
+      var responseObj = {
+        statusCode: 200,
+        body: {
+          data: {
+            token: "zaqwsxcderfv.qawsedrftg.qxderfvbhy"
+          }
+        }
+      };
+      AWS.mock("DynamoDB.DocumentClient", "query", (param, cb) => {
+        var dataObj = {
+          Items: [event.body]
+        }
+        return cb(null, dataObj);
+      });
+      var i=0;
+      reqStub = sinon.stub(request, "Request", (obj) => {
+        i++;
+        console.log(i)
+        if(obj.method.toLowerCase() === 'post') {
```

Comment:
same comment here. please don't do if/else. let the test do one thing only. please update in other tests as well

Reply:
Worked as per comments